src/main/java/org/brapi/test/BrAPITestServer/service/germ/GermplasmService.java

```diff
@@ -161,9 +151,164 @@ public Page<GermplasmEntity> findGermplasmEntities(@Valid GermplasmSearchRequest
 				.appendList(request.getFamilyCodes(), "familyCode");
 
 		Page<GermplasmEntity> page = germplasmRepository.findAllBySearch(searchQuery, pageReq);
+
+		if(!page.isEmpty()) {
+			log.debug("fetching xrefs");
+			fetchXrefs(page);
+			log.debug("fetching additionalInfo");
+			fetchAdditionalInfo(page);
+			log.debug("fetching attributes");
+			fetchAttributes(page);
+			log.debug("fetching donors");
+			fetchDonors(page);
+			log.debug("fetching origins");
+			fetchOrigin(page);
+			log.debug("fetching institutes");
+			fetchInstitutes(page);
+			log.debug("fetching taxons");
+			fetchTaxons(page);
+			log.debug("fetching storage codes");
+			fetchStorageCodes(page);
+			log.debug("fetching pedigree edges");
+			fetchPedigreeEdges(page);
+		}
+
 		return page;
 	}
 
+	private void fetchXrefs(Page<GermplasmEntity> page) {
+		SearchQueryBuilder<GermplasmEntity> searchQuery = new SearchQueryBuilder<GermplasmEntity>(GermplasmEntity.class);
+		searchQuery.leftJoinFetch("externalReferences", "externalReferences")
+				.leftJoinFetch("pedigree", "pedigree")
```

Comment:
Why is the LEFT JOIN FETCH with pedigree needed?

Reply:
Not entirely sure...when it wasn't there, then Hibernate was executing a select query against the pedigree table for each object in the result set.  Adding this leftJoinFetch resolved that.